### PR TITLE
Upgrade to flask-restx

### LINKED ISSUE #579: Upgrade to flask-restx
**Describe the bug**

`flask-restplus` is not maintained anymore, it's best we move to [flask-restx](https://github.com/python-restx/flask-restx)

A lot of modules break on newer Python versions because of `flask-restplus` staying unmaintained.

This is a big change, but it's a change in the right direction.

Relevant links:
https://github.com/noirbizarre/flask-restplus/issues/593

### PR DESCRIPTION
**Pull request by SanketDG**
### Note: This is a breaking change.

This PR marks the upgrade of `flask-restx` as discussed in #579, the framework that we use i.e. `flask_restplus 

Fixes #579

### Type of Change:

- Code

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged
- [x] Update Postman API at /docs folder
- [x] Update Swagger documentation and the exported file at /docs folder
- [x] Update requirements.txt

**Code/Quality Assurance Only**

- [x] My changes generate no new warnings 
- [x] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by rpattath** [comment]
@SanketDG I have removed Needs Testing label for now until the code is reviewed.

**Review by satya7289 — DISMISSED** [review]

**Comment by SanketDG** [comment]
Can this get some reviews? I will resolve the merge conflicts soon.

File: app/api/dao/user.py [hunk 1]
```diff
@@ -102,65 +102,69 @@ def delete_user(user_id: int):
...
     @staticmethod
     def list_users(user_id: int, search_query: str = "", page: int = DEFAULT_PAGE, per_page: int = DEFAULT_USERS_PER_PAGE, is_verified = None):
         """ Retrieves a list of verified users with the specified ID.
-        
+
         Arguments:
             user_id: The ID of the user to be listed.
             search_query: The search query for name of the users to be found.
             is_verified: Status of the user's verification; None when provided as an argument.
+<<<<<<< HEAD
```
Comment (isabelcosta) [inline app/api/dao/user.py]: @SanketDG 🙈

**Review by isabelcosta — APPROVED** [review]
Coding wise looks great! Thank you for doing this migration to flask-restx 🙌 @SanketDG 
I need one more approval and a QA person to do a simple test to make sure everything is working as expected manually.

**Comment by rpattath** [comment]
I removed `Needs Testing Label` because the PR does not have 2 approvals yet.

**Review by ramitsawhney27 — APPROVED** [review]
LGTM

**Review by foongminwong — APPROVED** [review]
Tested locally, no error.

**Comment by isabelcosta** [comment]
@foongminwong @ramitsawhney27 thank you so much! I will move forward and merge this.

**Pull request merged**